GPU/Common/TextureCacheCommon.cpp

```diff
@@ -1366,26 +1371,88 @@ static void DecodeDXTBlock(uint8_t *out, int outPitch, uint32_t texaddr, const u
 		h = (((int)limited / sizeof(DXTBlock)) / (bufw / 4)) * 4;
 	}
 
+	*fullAlphaMask = 1;  // we just use one bit here.
+
+	u32 alpha = 0xFFFFFFFF;
 	for (int y = 0; y < h; y += 4) {
 		u32 blockIndex = (y / 4) * (bufw / 4);
 		int blockHeight = std::min(h - y, 4);
 		for (int x = 0; x < minw; x += 4) {
-			if (n == 1)
-				DecodeDXT1Block(dst + outPitch32 * y + x, (const DXT1Block *)src + blockIndex, outPitch32, blockHeight, false);
-			if (n == 3)
+			switch (n) {
+			case 1:
+				DecodeDXT1Block(dst + outPitch32 * y + x, (const DXT1Block *)src + blockIndex, outPitch32, blockHeight, &alpha);
+				break;
+			case 3:
 				DecodeDXT3Block(dst + outPitch32 * y + x, (const DXT3Block *)src + blockIndex, outPitch32, blockHeight);
-			if (n == 5)
+				break;
+			case 5:
 				DecodeDXT5Block(dst + outPitch32 * y + x, (const DXT5Block *)src + blockIndex, outPitch32, blockHeight);
+				break;
+			}
 			blockIndex++;
 		}
 	}
+
+	switch (n) {
+	case 1:
+		*alphaSum = alpha;
+		break;
+	case 3:
+	case 5:
+		// Just report that we don't have full alpha, since these formats are made for that.
+		*alphaSum = 0;
+		break;
+	}
+
 	w = (w + 3) & ~3;
```

Comment:
This line happened in: 1f6860d6c16f454b06829fc93483440916cc3cf0, but `w` is unused after this. Not sure if that's a problem or just a useless line tho' :)

Reply:
Looks to me like the line is unnecessary indeed.

Reply:
It's just hung around since here:
https://github.com/hrydgard/ppsspp/commit/5fb8e6dfb07ad11efa7ec91ec2e632d03825e34b#diff-06771fb6126492de2d06dd342494acf73e3206bf2ef7d15bda486c9365cb5944R847

-[Unknown]